qualtran/bloqs/arithmetic/addition.py

```diff
@@ -69,15 +73,26 @@ class Add(GateWithRegisters, cirq.ArithmeticGate):
     def signature(self):
         return Signature.build(a=self.bitsize, b=self.bitsize)
 
-    def registers(self) -> Sequence[Union[int, Sequence[int]]]:
-        return [2] * self.bitsize, [2] * self.bitsize
+    def add_my_tensors(
+        self,
+        tn: 'qtn.TensorNetwork',
+        tag: Any,
+        *,
+        incoming: Dict[str, 'SoquetT'],
+        outgoing: Dict[str, 'SoquetT'],
+    ):
+        import quimb.tensor as qtn
+
+        N = 2**self.bitsize
+        inds = (incoming['a'], incoming['b'], outgoing['a'], outgoing['b'])
+        unitary = np.zeros((N,) * len(inds), dtype=np.complex128)
+        for a, b in itertools.product(range(N), range(N)):
+            unitary[a, b, a, (a + b) % N] = 1
```

Comment:
Why is this % N

Reply:
otherwise it would be out of range

Reply:
Do you know what happens with overflow? I'm sort of curious.

Reply:
The tests are written to never trigger overflow, which is something to say. 

looking at `cirq.ArithmeticGate` line 219 seems to imply that `% N` is used, which was the previous behavior.